readthedocs/core/utils/contact.py

```diff
@@ -67,41 +66,33 @@ class TempNotification(SiteNotification):
             success_level = message_constants.SUCCESS_PERSISTENT
 
         def render(self, *args, **kwargs):
-            context = {
-                'user': self.user,
-                'domain': f'https://{settings.PRODUCTION_DOMAIN}',
-            }
-            context.update(context_function(self.user))
             return markdown.markdown(
-                notification_template.render(Context(context))
+                notification_template.render(Context(self.get_context_data()))
             )
 
     total = users.count()
     for count, user in enumerate(users.iterator(), start=1):
         context = {
-            'user': user,
-            'domain': f'https://{settings.PRODUCTION_DOMAIN}',
+            "user": user,
+            "production_uri": f"https://{settings.PRODUCTION_DOMAIN}",
         }
         context.update(context_function(user))
 
         if notification_content:
             notification = TempNotification(
                 user=user,
                 success=True,
+                extra_context=context,
             )
             try:
                 if not dryrun:
                     backend.send(notification)
-                else:
-                    pprint(markdown.markdown(
-                        notification_template.render(Context(context))
-                    ))
```

Comment:
We should at least have an indication that we hit this code while running in dry mode, otherwise it isn't easy to see the effect that an operation will have. Same for the other dryrun check.

Reply:
It currently shows the `log.info` from the `else` clause.

Reply:
Calling render is a way to test the action itself without sending the notification, if there is an error in the template, we can catch it early.

Reply:
OK. That makes sense. I can add the render 👍

I didn't like printing it in the console because it's too verbose and it does it for every user. So, I'll remove the print

Reply:
I used `log.debug` for this.